scrapy/core/spidermw.py

```diff
@@ -60,8 +60,18 @@ def process_spider_exception(_failure):
             return _failure
 
         def process_spider_output(result):
+            def wrapper(result_iterable):
+                try:
+                    for r in result_iterable:
+                        yield r
+                except Exception as ex:
+                    exception_result = process_spider_exception(Failure(ex))
```

Comment:
Docs for [process_spider_exception](https://doc.scrapy.org/en/latest/topics/spider-middleware.html#scrapy.spidermiddlewares.SpiderMiddleware.process_spider_exception) method say:

> This method is called when when a spider or process_spider_input() method (from other spider middleware) raises an exception.

After this change method starts to fire when `process_spider_output` of a previous middleware raises an error; I'm not sure what are the implications, but it should be documented, and it could be backwards incompatible.